include/alpaka/ctx/block/CtxBlockOacc.hpp

```diff
@@ -0,0 +1,302 @@
+/* Copyright 2020 Jeffrey Kelling
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#ifdef ALPAKA_ACC_ANY_BT_OACC_ENABLED
+
+#if _OPENACC < 201306
+    #error If ALPAKA_ACC_ANY_BT_OACC_ENABLED is set, the compiler has to support OpenACC xx or higher!
+#endif
+
+// Base classes.
+#include <alpaka/workdiv/WorkDivMembers.hpp>
+#include <alpaka/idx/gb/IdxGbOacc.hpp>
+#include <alpaka/block/shared/dyn/BlockSharedMemDynMember.hpp>
+#include <alpaka/block/shared/st/BlockSharedMemStMember.hpp>
+#include <alpaka/block/sync/BlockSyncBarrierOacc.hpp>
+
+// Specialized traits.
+#include <alpaka/idx/Traits.hpp>
+
+#include <limits>
+#include <typeinfo>
+
+namespace alpaka
+{
+    template<
+        typename TDim,
+        typename TIdx,
+        typename TKernelFnObj,
+        typename... TArgs>
+    class TaskKernelOacc;
+
+    //#############################################################################
+    //! The OpenACC block context.
+    template<
+        typename TDim,
+        typename TIdx>
+    class CtxBlockOacc final :
+        public WorkDivMembers<TDim, TIdx>,
+        public gb::IdxGbOacc<TDim, TIdx>::BlockShared,
+        public BlockSharedMemDynMember<>,
+        public detail::BlockSharedMemStMemberImpl<4>,
+        public BlockSyncBarrierOacc::BlockShared,
```

Comment:
I do not see any need for nesting `BlockShared` within `BlockSyncBarrierOacc` here as well. Can't they get merged?